batch/batch/worker/worker.py

```diff
@@ -408,33 +415,39 @@ def __init__(self, job, name, spec):
         self.overlay_path = None
 
         self.image_config = None
+        self.image_digest = None
         self.rootfs_path = None
         scratch = self.spec['scratch']
         self.container_scratch = f'{scratch}/{self.name}'
         self.container_overlay_path = f'{self.container_scratch}/rootfs_overlay'
         self.config_path = f'{self.container_scratch}/config'
 
+        self.container_name = f'batch-{self.job.batch_id}-job-{self.job.job_id}-{self.name}'
+
         self.netns: Optional[NetworkNamespace] = None
         self.process = None
 
-    async def run(self, worker):
+    async def run(self, worker: 'Worker'):
         try:
 
             async def localize_rootfs():
-                async with worker.rootfs_locks[self.image_ref_str]:
+                cond = worker.rootfs_locks[self.image_ref_str]
+                async with cond:
+                    if len(worker.rootfs_digest_refs) == 10:
+                        unused_image = await cond.wait_for(lambda: get_unused_image(worker))
+                        shutil.rmtree(f'/host/rootfs/{unused_image}')
+                        del worker.rootfs_digest_refs[unused_image]
```

Comment:
Do you need to also delete the lock/condition for the unused image?

Reply:
This should delete the entry as well